rclpy/rclpy/executors.py

```diff
@@ -366,11 +367,35 @@ def spin_once_until_future_complete(
     def _take_timer(self, tmr):
         try:
             with tmr.handle:
-                tmr.handle.call_timer()
+                info = tmr.handle.call_timer_with_info()
+                timer_info = TimerInfo(
+                    expected_call_time=info['expected_call_time'],
+                    actual_call_time=info['actual_call_time'],
+                    clock_type=tmr.clock.clock_type)
+
+                # This needs to be done somewhere else since this is certain overhead to call timer
+                # for every iteration. at this moment stays here for review purpose.
+                # Besides, probably we can make this generic utility function
+                def check_argument_type(callback_func, target_type):
```

Comment:
I believe user can change the callback in the application, so signature check needs to be done here. but we can do that in the 1st place in this scope, so that we can save overhead especially for the application does not need `TimerInfo`.